eva_submission/evapro/populate_evapro.py

```diff
@@ -0,0 +1,454 @@
+        $sth_eva = $dbh_eva->prepare("insert into PROJECT_ENA_SUBMISSION (PROJECT_ACCESSION,SUBMISSION_ID) values (?,?)");
+        foreach my $sub_id(@{$accession_to_id->{'SUBMISSION_ID'}}){
+            $sth_eva->execute($project_accession,$sub_id);
+        }
+        $sth_eva = $dbh_eva->prepare("insert into EVA_SUBMISSION (EVA_SUBMISSION_ID,EVA_SUBMISSION_STATUS_ID) values (?,?)");
+        $sth_eva->execute($eload,6);
+        $sth_eva = $dbh_eva->prepare("insert into PROJECT_EVA_SUBMISSION (PROJECT_ACCESSION,old_ticket_id,ELOAD_ID) values (?,?,?)");
+        $sth_eva->execute($project_accession,$eload,$eload);
+        '''
+        ProjectEnaSubmission(project_accession=project_obj.project_accession, submission_id=submission_obj.submission_id)
+        EvaSubmission(eva_submission_id=eload, eva_submission_status_id=6)
+        ProjectEvaSubmission(project_accession=project_obj.project_accession, old_ticket_id=eload, eload_id=eload)
+
+
+    def insert_analysis(self, analysis_accession, title, alias, description, center_name, date, assembly_set_id):
+        query = select(Analysis).where(Analysis.analysis_accession == analysis_accession)
+        result = self.eva_session.execute(query).fetchone()
+        if result:
+            analysis_obj = result.Analysis
+        else:
+            analysis_obj = Analysis(analysis_accession=analysis_accession, title=title, alias=alias, description=description,
+                     center_name=center_name, date=date, assembly_set_id=assembly_set_id)
+            self.eva_session.add(analysis_obj)
+        return analysis_obj
+
+    def insert_assembly_set(self, taxonomy_obj, assembly_accession, assembly_name=None):
+        if not assembly_name:
+            assembly_name = get_ncbi_assembly_name_from_term(assembly_accession, api_key=cfg.get('eutils_api_key'))
+
+        query = select(AssemblySet).where(AssemblySet.taxonomy_id == taxonomy_obj.taxonomy_id,
+                                          AssemblySet.assembly_name == assembly_name)
+        result = self.eva_session.execute(query).fetchone()
+        if result:
+            assembly_set_obj = result.AssemblySet
+        else:
+            assembly_code = self.get_assembly_code(assembly_accession, ncbi_api_key=cfg.get('eutils_api_key'))
+            assembly_set_obj = AssemblySet(taxonomy_id=taxonomy_obj.taxonomy_id, assembly_name=assembly_name, assembly_code=assembly_code)
+            self.eva_session.add(assembly_set_obj)
+            self.eva_session.flush()
+
+            accessioned_assembly_obj = AccessionedAssembly(
+                assembly_set_id=assembly_set_obj.assembly_set_id, assembly_accession=assembly_accession,
+                assembly_chain=assembly_accession.split('.')[0], assembly_version=assembly_accession.split('.')[1]
+            )
+            self.eva_session.add(accessioned_assembly_obj)
+
+        return assembly_set_obj
+
+    def insert_custom_assembly_set(self, taxonomy_obj, custom_assembly):
+        '''
```

Comment:
I guess this is a `TODO`?
Incidentally I guess we might have to think about how to combine this with anything we re-implemented in eva-submission, e.g. [here](https://github.com/EBIvariation/eva-submission/blob/master/eva_submission/eload_ingestion.py#L595)